### PR TITLE
[FIX] sale_project: avoid opening form when creating SOLs

### PR DESCRIPTION
**Pull request by MissingNoShiny**
When creating a SOL on the fly, if the name entered matches an existing service product, selecting 'Create' should create the SOL without opening a form view modal. We do that by fetching the `default_name` from the context. However, from the Many2one field, since the name will already be present in the create vals, in will not be in the default values in the context.

To fix this, this PR adds a new context key when calling `name_create` that allows us to retrieve the name from the `default_get`

Enterprise: https://github.com/odoo/enterprise/pull/53833

Task-3553151

### CODE REVIEW

**Review by pkyriakou — COMMENTED** [review]
Thank you @MissingNoShiny for your work :100: 
Nothing really to comment on :raised_hands: LGTM :green_circle:

File: addons/sale_project/models/sale_order_line.py [hunk 1]
```diff
@@ -58,16 +58,43 @@ def default_get(self, fields):
...
+    @api.model
+    def _add_missing_default_values(self, values):
+        # When creating a SOL through the quick create, the name_create will be
+        # called with whatever was typed in the field. However, we don't want
+        # that value to overwrite the computed SOL name if we find a product.
+        defaults = super()._add_missing_default_values(values)
+        if self.env.context.get('form_view_ref') == 'sale_project.sale_order_line_view_form_editable':
+            if "name" in defaults and "product_id" in defaults:
+                del defaults["name"]
+        return defaults
```
Comment (MissingNoShiny) [inline addons/sale_project/models/sale_order_line.py:96]: I shouldn't be overriding this, but I don't know how else I could avoid the name typed in the field from being used in the `create`.

**Comment by MissingNoShiny** [comment]
The code required for this feature keeps getting worse :pensive:

File: addons/sale_project/models/sale_order_line.py [hunk 2]
```diff
@@ -118,7 +142,7 @@ def create(self, vals_list):
                     line.order_id.message_post(body=msg_body)
 
         # Set a service SOL on the project, if any is given
-        if (project_id := self.env.context.get('link_to_project')):
+        if project_id := self.env.context.get('link_to_project'):
```
Comment (pkyriakou) [inline addons/sale_project/models/sale_order_line.py:145]: I guess this is an unnecessary change here :thinking:

**Review by pkyriakou — COMMENTED** [review]
Thank you @MissingNoShiny for your updates and work :100: 
I dont really have anything to comment apart from that little tiny thing. I agree that the solution is convoluted but I don't personally see another way of achieving this. Good work with the comments, at least we will know what is going on :sweat_smile: 
I will be greening the task regardless :v:

**Review by xavierbol — COMMENTED** [review]
Thanks for your work. 🙂 

Before merging your fixs, is it possible to add a test for the changes you made in the last commit? 🤔 

If yes, can you do it? 🥺 

Thanks in advance! 🙂

**Review by xavierbol — APPROVED** [review]
LGTM, thanks again for your work. 🙂 

robodoo rebase-ff r+
